### ISSUE TITLE
Develop an apsim file runner/server tool

### ISSUE THREAD

**Issue by hol430** [APSIMInitiative/ApsimX#6529]
Running a parameter optimisation/sensitivity analysis typically requires an .apsimx file to be run many times serially. Currently this can be done only by repeatedly invoking Models.exe, with all of the overhead that entails (disk IO, model deserialization, connection of links/events, factor initialisation, etc). The proposed solution to this problem is to implement a "server" application which reads an .apsimx file once, and performs all of this initialisation once. It will then hold the simulations in memory, ready to be run on-demand with minimal overheads.

**Comment by lie112** [comment]
Is this also applicable to the experiment with factors simulations, or is it the same thing to perform sensitivity analysis?

**Comment by hol430** [comment]
This will work fine if you have an experiment in your file - it will rerun the whole file each time it receives a "run" command.

**Comment by BrianCollinss** [comment]
@hol430 Hi Drew. What does the user need to do to be able to use this server thing? Does APSIM dceicde when to use this when it faces many simulations?

**Comment by BrianCollinss** [comment]
Thanks. That's what I meant, e.g. running a factorial experiment with one simulation as the base and changing the parameters many times. Is there any instruction on how to use this?

**Comment by hol430** [comment]
Yeah, there are some instructions [here](https://apsimnextgeneration.netlify.app/usage/server/). The command-line arguments section may be slightly out of date, I'm not sure. Running with `--help` should give you a full list of options with help text, otherwise feel free to ask here if anything doesn't make sense.
